### DISCUSSION TITLE
iPxe stuck initializing Surface Tablets and Lenovo laptops

### DISCUSSION THREAD

**Discussion by ShaowulfD** [ipxe/ipxe#1360] [General]
Hello!

I need some assistance in determining the cause of an issue that is affecting our organisation's ability to image systems

We are using the Quest KACE Systems Deployment Appliance that is using iPxe to boot into the deployment environment

The Version if iPxe.efi we are using is  1.21.1+  I am not certain if that is the most up to date version or not and am unsure how to check

Recently the Surface Pro 7 and Surface Pro 9 as well as the Lenovo X1 Carbon Gen 10 and Gen 11 received a firmware upgrade and since that upgrade we can no longer boot into our deployment environment, it downloads the iPXE.efi file and then gets stuck at the Initializing Device stage.

I have opened a ticket with Quest but its not moving very quickly and logging the ability to inplace image our devices is causing quite a bit of issues for us.  

I have seen others reporting some similar issues on here but with different environments and  Iam not certain if our environment matters in this case.

Any assistance or advice would be greatly appreciated!, I am happy to provide additional details if needed!

![Surface9_PXE Error](https://github.com/user-attachments/assets/e5e33495-3930-4b39-bc11-8950a3fadac7)

**Comment by NiKiZe** [comment]
Please provide the exact version of iPXE that is used, also try with snponly.efi

**Reply by ShaowulfD** [reply]
I reconfigured DHCP to use snponly.efi however it did not seem to make a difference

KACE does not allow access to the internal file system so  I cannot access ipxe.efi directly however here is a pre-firmware upgrade screenshot hopefully it has the info needed!

![LenovoX1CarbonGen11_PXEBootjpg](https://github.com/user-attachments/assets/99efca5a-4aee-47fe-a688-b0eb0d187dfb)

**Reply by NiKiZe** [reply]
https://github.com/ipxe/ipxe/commit/7099 does not exist.
By the license, if you request the sources, they have to give it to you.... For iPXE there is special license where they dont have to provide sources of they are using an unmodified version of iPXE. Since the commit they are using is not known, it clearly isn't unmodified, it might just be config changes or similar, but they have to show it.

**Reply by ShaowulfD** [reply]
So if I’m understanding what you’re telling me the show version of ipxe.efi in my screenshot is not one that exists.

I will communicate with the folks at quest to see if they can provide me with the exact version they are using

Can you tell me what the most up-to-date version is so I can give that to them as a reference?

**Reply by NiKiZe** [reply]
The g is followed by the git commit hash, you can check current master for what the latest is.

It is always a good idea to try latest ipxe and see if that boots on the hardware, but in this case I think you will have to find a version that does work and compare that, and maybe even bisecting to find what failed... In this case it would bee good, as you mentioned, to know how old the iPXE version is, to then figure out what the best path forward is.

**Reply by ShaowulfD** [reply]
Thank you for the responses!

I have put the question to Quest to provide the detailed version information and have referenced and linked this thread for them and we can go from there hopefully updating the file if it is not the most recent version will have a positive effect!

**Reply by NiKiZe** [reply]
I did reach out to quest as well, they do claim official unmodified source,veven if we do know something differs, but at least they got back with this.
> They posted a copy of the (unmodified) iPXE source code on https://opensource.quest.com/ , and the component is included in the acknowledgement section in the next release.
